src/components/file-browser/file-browser-hooks.test.ts

```diff
@@ -0,0 +1,110 @@
+import { act, renderHook } from '@testing-library/react-hooks'
+import _ from 'lodash/fp'
+import { IncrementalResponse } from 'src/components/file-browser/file-browser-backends'
+import { useIncrementalResponse } from 'src/components/file-browser/file-browser-hooks'
+
+
+describe('useIncrementalResponse', () => {
+  // Returns an incremental response with 3 pages of 3 numbers each
+  const getTestIncrementalResponse = (): Promise<IncrementalResponse<number>> => {
+    const getNextPage = (previousResults: number[], pageNumber: number): Promise<IncrementalResponse<number>> => {
+      const results = [...previousResults, ..._.range(pageNumber * 3 + 1, (pageNumber + 1) * 3 + 1)]
+      const hasNextPage = pageNumber < 2
+      return Promise.resolve({
+        results,
+        getNextPage: hasNextPage ?
+          () => getNextPage(results, pageNumber + 1) :
+          () => { throw new Error('No next page') },
+        hasNextPage
+      })
+    }
+
+    const firstPageResults = [1, 2, 3]
+    return Promise.resolve({
+      results: firstPageResults,
+      getNextPage: () => getNextPage(firstPageResults, 1),
+      hasNextPage: true
+    })
+  }
+
+  it('gets initial response', async () => {
+    const { result, waitForNextUpdate } = renderHook(() => useIncrementalResponse(getTestIncrementalResponse))
```

Comment:
please add AAA section comments to these unit tests:
// Arrange
// Act
// Assert

Reply:
Done in https://github.com/DataBiosphere/terra-ui/pull/3468/commits/fa6803cb9a3750c21dd3a1d0f9a57750e229ddfa.